### PR TITLE
feat(episodes-timline): 11912 add episodes timeline feature

### PR DESCRIPTION
**Pull request by Akiyamka**
image

### CODE REVIEW

**Comment by Komzpa** [comment]
an image is worth a thousand words

File: src/features/event_episodes/components/EpisodesTimeline/EpisodesTimeline.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
+import { Timeline } from '@konturio/ui-kit';
+import { useAtom } from '@reatom/react';
+import { useCallback, useMemo } from 'react';
+import { eventEpisodesController } from '../../controller';
+import { eventEpisodesModel } from '../../model';
+import type { TimelineProps } from '@konturio/ui-kit';
+import type { Episode } from '~core/types';
+
+type WithRequired<T, K extends keyof T> = T & { [P in K]-?: T[P] };
```
Comment (propakov) [inline src/features/event_episodes/components/EpisodesTimeline/EpisodesTimeline.tsx:9]: `-?` would be awesome to get some explanation or link 😄

**Review by propakov — APPROVED** [review]
👍

File: src/features/event_episodes/components/EpisodesTimeline/EpisodesTimeline.tsx (hunk 1 above)
Comment (Akiyamka) [reply src/features/event_episodes/components/EpisodesTimeline/EpisodesTimeline.tsx:9]: this is literally what it looks like, it removes the optionality from the parameter (remove "?")
https://www.typescriptlang.org/docs/handbook/2/mapped-types.html#mapping-modifiers

Comment (propakov) [reply src/features/event_episodes/components/EpisodesTimeline/EpisodesTimeline.tsx:9]: wow, cool! thanks for explanation 🙏

File: src/utils/atoms/createController.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      }
+    };
+    return acc;
+  }, {} as T);
+}
+
+const example = createFeatureController({
+  foo: () => null,
+  bar: (a: string) => a,
+});
```
Comment (IlyaIzr) [inline src/utils/atoms/createController.ts:22]: should be deleted?

**Pull request merged**